Sources/AppStoreConnectCLI/Services/EndpointRequestor.swift

```diff
@@ -24,3 +24,24 @@ struct DefaultEndpointRequestor: EndpointRequestor {
         }
     }
 }
+
+protocol PaginatedResponse: Decodable {
+    var links: PagedDocumentLinks { get }
+}
+
+extension EndpointRequestor {
+    func requestAllPages<T: PaginatedResponse>(
+        with endpointProvider: @escaping (PagedDocumentLinks?) -> APIEndpoint<T>,
+        next: PagedDocumentLinks? = nil
+    ) -> AnyPublisher<[T], Error> {
+        request(endpointProvider(next))
+            .flatMap { (response) -> AnyPublisher<[T], Error> in
+                response.links.next != nil
+                    ? self.requestAllPages(with: endpointProvider, next: response.links)
+                        .flatMap { Empty<[T], Error>().append([response] + $0) }
+                        .eraseToAnyPublisher()
+                    : Empty<[T], Error>().append([response]).eraseToAnyPublisher()
```

Comment:
Even though it's multi return a guard is probably clearer here rather than a large ternery

```suggestion
                guard response.links.next != nil else {
                    return Empty<[T], Error>().append([response]).eraseToAnyPublisher()
                }
                
                return self.requestAllPages(with: endpointProvider, next: response.links)
                    .flatMap { Empty<[T], Error>().append([response] + $0) }
                    .eraseToAnyPublisher()
```

What do you think?